posts/2024/2024_07_06_Inigo_week_6.rst

```diff
@@ -0,0 +1,48 @@
+Week 6 into GSoC 2024: Stuck with the Variational AutoEncoder, problems with Keras
+==================================================================================
+
+
+.. post:: July 06 2024
+    :author: Iñigo Tellaetxe
+    :tags: google
+    :category: gsoc
+
+
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+
+This week was all about the Variational AutoEncoder. My mentors advised me to drop the TensorFlow implementation of the regression VAE I found last week, to instead directly integrate the variational and conditional characteristics in my AE implementation, following a more modular approach. This was a good decision, as adapting third party code to one's needs is often a bit of a mess (it already started being a mess, so yeah). Also, once the variational part is done, implementing the conditional should not be that hard.
+
+To provide a bit of intuition behind the VAE, let me illustrate first the "vanilla" AE. This is a neural network that compresses the input data into a reduced representation, and then tries to reconstruct the original data from this representation. We refer to "latent space" to the place where the compressed data representations live. So, once the AE is good at compressing data to the latent space, we could take a sample from it and generate new data.
+The objective of the AE is to minimize the difference between the input data and the generated data, also called the "reconstruction loss".
+
+.. image:: /_static/images/inigo_vanilla_autoencoder.png
+    :alt: Vanilla AutoEncoder diagram.
+    :width: 600
+
+
+Usually the problem with AEs is that the latent space is full of "holes", i.e. it is a set of points, meaning that if a sample is taken between two already encoded samples, the generated data will not be an interpolation between the two samples, but a random sample. To solve this, the VAE seeks to "regularize" the latent space, encoding/compressing the input data into distributions that live in it, instead of single points. This way, the latent space is continuous and the interpolation between two samples is meaningful and more likely to follow the data distribution.
```

Comment:
No need to change this, but bear in mind that as you say, the latent space that we will get will have a normal distribution, across all bundles: so one single normal across all bundles. Some time ago I think I shared a paper where they showed the type of tractography reconstructions that they got using AEs, VAEs, VQ-AEs, etc.